src/main/java/ca/corefacility/bioinformatics/irida/processing/impl/AutomatedAnalysisFileProcessor.java

```diff
@@ -0,0 +1,156 @@
+
+	private final SampleSequencingObjectJoinRepository ssoRepository;
+	private final ProjectSampleJoinRepository psjRepository;
+	private final AnalysisSubmissionRepository submissionRepository;
+	private final AnalysisSubmissionTemplateRepository analysisTemplateRepository;
+	private final IridaWorkflowsService workflowsService;
+	private final SequencingObjectRepository objectRepository;
+
+	@Autowired
+	public AutomatedAnalysisFileProcessor(SampleSequencingObjectJoinRepository ssoRepository,
+			ProjectSampleJoinRepository psjRepository, AnalysisSubmissionRepository submissionRepository,
+			AnalysisSubmissionTemplateRepository analysisTemplateRepository, IridaWorkflowsService workflowsService,
+			SequencingObjectRepository objectRepository) {
+		this.ssoRepository = ssoRepository;
+		this.psjRepository = psjRepository;
+		this.submissionRepository = submissionRepository;
+		this.analysisTemplateRepository = analysisTemplateRepository;
+		this.workflowsService = workflowsService;
+		this.objectRepository = objectRepository;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void process(SequencingObject sequencingObject) {
+		List<AnalysisSubmissionTemplate> analysisTemplates = getAnalysisTemplates(sequencingObject);
+
+		for (AnalysisSubmissionTemplate template : analysisTemplates) {
+
+			// build an SubmittableAnalysisSubmission
+			AnalysisSubmission.Builder builder = new AnalysisSubmission.Builder(template);
+
+			AnalysisSubmission submission = builder.inputFiles(Sets.newHashSet(sequencingObject))
+					.build();
+			submission.setSubmitter(template.getSubmitter());
+
+			submission = submissionRepository.save(submission);
+
+			legacyFileProcessorCompatibility(submission, sequencingObject);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean modifiesFile() {
+		return false;
+	}
+
+	/**
+	 * Get the {@link AnalysisSubmissionTemplate}s for a given {@link SequencingObject}.  Search up the {@link Sample}
+	 * and {@link Project}s the sequence belongs to.
+	 *
+	 * @param object the {@link SequencingObject}
+	 * @return the List of {@link AnalysisSubmissionTemplate}
+	 */
+	private List<AnalysisSubmissionTemplate> getAnalysisTemplates(SequencingObject object) {
+		List<AnalysisSubmissionTemplate> submissionTemplates = new ArrayList<>();
+
+		SampleSequencingObjectJoin sampleForSequencingObject = ssoRepository.getSampleForSequencingObject(object);
+
+		/*
+		 * This is something that should only ever happen in tests, but added
+		 * check with a warning
+		 */
+		if (sampleForSequencingObject != null) {
+			List<Join<Project, Sample>> projectForSample = psjRepository.getProjectForSample(
+					sampleForSequencingObject.getSubject());
+
+			for (Join<Project, Sample> j : projectForSample) {
+				List<AnalysisSubmissionTemplate> analysisSubmissionTemplatesForProject = analysisTemplateRepository.getAnalysisSubmissionTemplatesForProject(
```

Comment:
Thinking about this now though, there still can be conflicts. Imagine this scenario:

* Two projects `A` and `B` each with an automated SISTR set up to run
* Sample `S` is shared among projects `A` and `B`
* Data gets uploaded to sample `S`, which projects SISTR pipeline should be associated with the sequencing object (project `A` or `B` SISTR results)?

We could just have it be in whichever SISTR results get completed last (which will be the default behaviour). Are there any conflicts you can think this could lead to?

Reply:
It's a similar problem we had in the past if multiple project were set to run SISTR or the assembly.  Just now since you can change params it could change results instead of just running redundantly.  I truly can't think of any way around this except just that the 2nd exec overwrites the 1st one.

Reply:
That sounds like a good solution to me right now. I'm not sure what else we would do either that's not overly complicated.